clusterloader2/pkg/util/error_list.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"strings"
+	"sync"
+)
+
+// ErrorList is a thread-safe error list.
+type ErrorList struct {
+	lock   sync.Mutex
```

Comment:
Maybe use RWMutex?

Reply:
I think it is not needed - this will mostly perform write operations.

Reply:
Yeah, that should be ok.